File: src/main/java/org/springframework/samples/petclinic/service/interfaces/VetService.java
package org.springframework.samples.petclinic.service.interfaces;

import org.springframework.samples.petclinic.dtos.ResponseData;
import org.springframework.samples.petclinic.model.Specialty;
import org.springframework.samples.petclinic.model.Vet;

import java.util.Collection;
import java.util.Set;


public interface VetService {

    ResponseData<Collection<Vet>> getVets();

    ResponseData<Vet> getVetById(int vetId);

    ResponseData<String> saveVet(Vet vet);

    ResponseData<Set<Specialty>> getSpecialties();

    ResponseData<String> addSpecialtyToVet(int vetId, Collection<Specialty> specialty);
}

File: client/src/services/vets/vetService.js
import clinicRequest from '../clinicRequest';

const endpoint = '/vet';

const getVets = () => {
  return clinicRequest.get(endpoint);
};

const saveVet = vet => {
  return clinicRequest.post(endpoint, vet);
};

const getVetSpecialties = () => {
  return clinicRequest.get(endpoint + '/specialty');
};

const getVetById = vetId => {
  return clinicRequest.get(endpoint + `/${vetId}`);
};

const vetService = { getVets, saveVet, getVetSpecialties, getVetById };
export default vetService;
